aioseop_class.php

```diff
@@ -4301,153 +4305,130 @@ function apply_description_format( $description, $post = null ) {
+				( is_tax() && in_array( get_query_var( 'taxonomy' ), $tax_noindex ) )
 		) {
-			$noindex = 'noindex';
+			$noindex = true;
+		}
 
-			// #322: duplicating this code so that we don't step on some other entities' toes.
-			if (
-					( 'on' === $aiosp_nofollow ) ||
-					(
-						( ! empty( $aioseop_options['aiosp_paginated_nofollow'] ) ) &&
-						$page > 1
-					) ||
-					(
-							( '' === $aiosp_nofollow ) &&
-							( ! empty( $aioseop_options['aiosp_cpostnofollow'] ) ) &&
-							in_array( $post_type, $aioseop_options['aiosp_cpostnofollow'] )
-					)
-			) {
-				$nofollow = 'nofollow';
-			}
-			// #322: duplicating this code so that we don't step on some other entities' toes.
-		} elseif (
+		if (
 				is_single() ||
 				is_page() ||
 				$this->is_static_posts_page() ||
 				is_attachment() ||
-				is_category() ||
-				is_tag() ||
-				is_tax() ||
-				( $page > 1 ) ||
 				$this->check_singular()
 		) {
-			$post_type = get_post_type();
+			if ( '' === $aiosp_noindex &&
+					! empty( $aioseop_options['aiosp_cpostnoindex'] ) &&
+					in_array( $post_type, $aioseop_options['aiosp_cpostnoindex'] )
+			) {
+				$noindex = true;
+			}
 			if (
-					$aiosp_noindex ||
-					$aiosp_nofollow ||
-					! empty( $aioseop_options['aiosp_cpostnoindex'] ) ||
-					! empty( $aioseop_options['aiosp_cpostnofollow'] ) ||
-					! empty( $aioseop_options['aiosp_paginated_noindex'] ) ||
-					! empty( $aioseop_options['aiosp_paginated_nofollow'] )
+					'' === $aiosp_nofollow &&
+					! empty( $aioseop_options['aiosp_cpostnofollow'] ) &&
+					in_array( $post_type, $aioseop_options['aiosp_cpostnofollow'] )
 			) {
-
-				if (
-						( 'on' === $aiosp_noindex ) ||
-						(
-							( ! empty( $aioseop_options['aiosp_paginated_noindex'] ) )
-							&& $page > 1
-						) ||
-						(
-							( '' === $aiosp_noindex ) &&
-							( ! empty( $aioseop_options['aiosp_cpostnoindex'] ) ) &&
-							in_array( $post_type, $aioseop_options['aiosp_cpostnoindex'] )
-						)
-
-				) {
-					$noindex = 'noindex';
-				}
-				if (
-						( $aiosp_nofollow == 'on' ) ||
-						(
-							( ! empty( $aioseop_options['aiosp_paginated_nofollow'] ) ) &&
-							$page > 1
-						) ||
-						(
-							( $aiosp_nofollow == '' ) &&
-							( ! empty( $aioseop_options['aiosp_cpostnofollow'] ) ) &&
-							in_array( $post_type, $aioseop_options['aiosp_cpostnofollow'] )
-						)
-				) {
-					$nofollow = 'nofollow';
-				}
+				$nofollow = true;
 			}
 		}
-		if ( is_singular() && $this->is_password_protected() && apply_filters( 'aiosp_noindex_password_posts', false ) ) {
-			$noindex = 'noindex';
+
+		return $this->aioseop_get_robots_meta_string( $noindex, $nofollow );
+	}
+
+	/**
+	 * The get_robots_meta_string() function.
+
+	 * Helper function for get_robots_meta().
+	 *
+	 * @since 3.2.0
+	 *
+	 * @param bool $noindex
+	 * @param bool $nofollow
+	 *
+	 * @return string
+	 */
+	private function aioseop_get_robots_meta_string( $noindex, $nofollow ) {
+		if ( $noindex ) {
```

Comment:
Sorry I didn't catch this sooner ( I thought AccessLint also looked for these type of patterns ), but one coding patterns that would improve this is.

```
$index_value  = 'index';
$follow_value = 'follow';

if ( $noindex ) {
	$index_value = 'noindex';
}

if ( $nofollow ) {
	$follow_value = 'nofollow';
}
```

Reply:
Definitely. Not sure why I didn't think of this.